### PR TITLE
Microcopy fix and helptext uniformity Attempt #2

### PR DESCRIPTION
**Pull request by NAretakis**
This is an attempt to fix the two errors that my previous PR created. I think I figured out where the missing translation problem was occurring in both the Activity Description section, as well as Expenses section.

I also tried to create some uniformity with each helptext and reminder field so that everything is linked to its corresponding section. For example: "sectiontitle" should now have a corresponding "sectiontitlehelptext" help text that goes along with it.

### This pull request changes...
- fixes the two missing translation errors
- standardizes the helptext and reminder inputs

### This pull request is ready to merge when...
- [ ] Tests have been updated (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author

### This feature is done when...
- [ ] One of the reviewers has tested the file on a local copy to make sure Product didn't break it again.
- [x] Design has approved the experience.

### CODE REVIEW

**Comment by quarterback** [comment]
@brendansudol @mgwalker Does changing "helptext" to "sectiontitlehelptext" etc. break anything? Also @NAretakis you didn't have the latest version so there are conflicts, but I can implement those changes you made so long as it doesn't cause any issues.

**Comment by NAretakis** [comment]
I went through and resolved the conflicts, so I think were good there. But please double check. Definitely need Devs to weigh in on changing "helptext" to "sectiontitlehelptext". The motive here was clarity for each bit of helptext but if it breaks things we can revert.

**Comment by quarterback** [comment]
It's good otherwise. I just wanted to make sure!

**Comment by NAretakis** [comment]
Ok, so let's figure this out before we merge this in. Is it cleaner to just use "helptext" & "reminder". And therefore not have to change the resource file? The reason I changed this was to give some clarity regarding which helptext and which reminders correspond to the various sections. As this PR stands it has:
- "sectiontitlehelptext"
- "titlehelptext"
- "titlereminder"
- "listtitlereminder"
- "alternativeshelptext"
- "subheaderhelptext"
- "subheaderreminder"
- "objectiveheaderhelptext"

**Comment by NAretakis** [comment]
if we do just use "helptext" and "reminder" I might need some help to figure out how we solve a duplicate error thingy that showed up when we had just the two things.

**Comment by mgwalker** [comment]
We initially tried to make the resource file "structured" so that you'd infer what `helpText` belonged to based on how it sat in the file.  E.g.:

```json
{
    "section": {
        "helpText": "this help text is for the section",
        "overview": {
            "helpText": "this help text is for the section, overview subsection"
        }
    }
}
```

If that's not helpful to you, or if using longer, more descriptive names is more helpful, we can definitely make the change.  But!  Any change needs to be done in concert between content and development.  🙂

Neither is cleaner or better from a dev perspective.  I'd defer to the folks actually writing the content, so you @NAretakis and @quarterback.

**Comment by NAretakis** [comment]
I think I actually like the simplicity of how it is structured, but the problem I ran into was multiple "helptext" fields, is there a way to address this in the code below?

```json
	"activities": {
		"title": "Program Activities",
		"helptext": "Please individually identify the activities referenced from the overview section. We'll collect information for each activity.",
		"listTitle": "Activity List",
		"helptext": "Examples of individual activities include: Auditing, Outreach, Environmental Scans, HIE Registry Development, CQM Database Development, Data and Analytics, HIE Onboarding, etc.",
		"reminder": "Please Note: The first activity in this list is Program Administration. Program Administration is where you should include information that pertains to implementing the EHR Incentive program. Examples of information that belongs within this activity includes: State and Contractor costs associated with operating the EHR incentive program and SLR Enhancements/Upgrades.",
		"addActivityButtonText": "Add activity",
		"deleteActivityButtonText": "Remove Activity",
		"noActivityMessage": "It looks like you don’t have any activities. Please add one below.",`
```

**Comment by mgwalker** [comment]
Ooooh, I see!  That's not the problem I thought it was.  One options is we could add more structure:

```json
"activities": {
  "title": "Program Activities",
  "helptext": "Please individually identify the activities referenced from the overview section. We'll collect information for each activity.",
  "list": {
    "title": "Activity List",
    "helptext": "Examples of individual activities include: Auditing, Outreach, Environmental Scans, HIE Registry Development, CQM Database Development, Data and Analytics, HIE Onboarding, etc.",
    "reminder": "Please Note: The first activity in this list is Program Administration. Program Administration is where you should include information that pertains to implementing the EHR Incentive program. Examples of information that belongs within this activity includes: State and Contractor costs associated with operating the EHR incentive program and SLR Enhancements/Upgrades.",
  },
  "addActivityButtonText": "Add activity",
  "deleteActivityButtonText": "Remove Activity",
  "noActivityMessage": "It looks like you don’t have any activities. Please add one below.",`
```

That would fit pretty well with the proposal I dropped into chat this morning about how we could have a consistent set of fields that apply to every section and subsection:

screen shot 2018-05-23 at 9 46 40 am

What do you think?

**Comment by NAretakis** [comment]
This looks good to me. If I take a stab at this, can you do the review and test/fix on a local copy?

**Comment by mgwalker** [comment]
Created #597 to capture the need to update how "reminder" text is displayed.  @lauraponce Should that be a design task before it goes to dev?

**Comment by mgwalker** [comment]
Going to assume that #594 will be merged, so I'll go ahead and get started moving the updated text y'all have done into the updated resource file.

**Comment by mgwalker** [comment]
Alright, they're merged.  Hopefully I didn't squash any of the content changes y'all made.  😬

**Comment by NAretakis** [comment]
@mgwalker ok I think I got my commits in now. Can you put on a local copy to test/fix before merging? And then I promise I'll stop breaking stuffs.

File: web/src/components/HelpText.js [hunk 1]
```diff
@@ -8,10 +8,12 @@ const HelpText = ({ text: textResource, reminder: reminderResource }) => {
...
-    <div className="mb2" style={{ whiteSpace: 'pre-line' }}>
-      {text && <div>{text}</div>}
-      {reminder && <div className="red">{reminder}</div>}
-    </div>
+    (text || reminder) && (
+      <div className="mb2" style={{ whiteSpace: 'pre-line' }}>
+        {text && <div>{text}</div>}
+        {reminder && <div className="red">{reminder}</div>}
+      </div>
+    )
```
Comment (mgwalker) [inline web/src/components/HelpText.js]: Makes the `<HelpText>` component empty if there's no help text or reminder text.  This way we can use it liberally and not worry about the text exists in the resource file, basically putting in placeholders that magically get filled in if the content need arises.

**Comment by brendansudol** [comment]
👍

**Pull request merged**